### ISSUE TITLE
Move from using unsound `from_utf8_unchecked` to safe `from_utf8` forwarding error

### ISSUE THREAD

**Issue by hashedone** [CosmWasm/cw-plus#393]
Discussion about this was part of previous PR: https://github.com/confio/tfi/pull/46#discussion_r692820745

TL;DR:
Overhead of verifying utf8 strings if very long, as it is `O(str.len())`, and we are typically operating on very short strings. Comparing to another overheads we introduce for sanity checks it is close to zero cost. The problem is, that if for any reason (not only attack, but also bugs) string assumed to be utf8 encoded ends up not to be, it using `unchecked` variant is not introducing crash - it introduces UB, and in worst case it might be exploitable. It doesn't make sense to abandon all safety guarantees of Rust just to avoid quick string scan. Also besides of being safest, the safe version is actually shorter, and doesn't involve `unsafe` block which is messy and always involves second thoughts while reading (or at least should).

**Comment by ueco-jb** [comment]
If my search is correct then I think there are no `from_utf8_unchecked` calls in `cw-plus` repository.

**Comment by ueco-jb** [comment]
To close with `not relevant` comment.

**Comment by maurolacy** [comment]
Closing this.
